File: Active/TypeScript/src/P010/P010.ts
// Project Euler
// Problem 10: The sum of all prime numbers below 2000000

import primeGenerator from "../lib/eratosthenesSieve";

export function compute010(limit: number): number {
    let sum: number = 0;
    // -1 because it asks all the sum 'below' limit, and the primeGenerator
    // generate prime from 2 to the limit (included)
    let num: number[] = primeGenerator(limit - 1);

    const len = num.length;
    for (let i = 0; i < len; i++) {
        sum += num[i];
    }

    return sum;
}

function run010(): void {
    const limit: number = 2_000_000;
    console.log(compute010(limit).toString());
}

// run010();

File: Active/Go/solutions/p010.go
// Project Euler
// Problem 10: The sum of all prime numbers below 2_000_000

package go_euler

import "go_euler/lib"

func Problem_010(limit int) int {
	numbers := go_euler.Eratosthenes_Sieve(limit)

	sum_of_primes := 0;
	for _, number := range numbers {
		sum_of_primes += number
	}

	return sum_of_primes
}
